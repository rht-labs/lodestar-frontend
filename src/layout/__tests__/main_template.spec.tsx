import React from 'react';
import {act, cleanup, fireEvent, queryByAttribute, render} from '@testing-library/react';
import {MainTemplate} from '../main_template';
import {MemoryRouter} from 'react-router';
import "@testing-library/jest-dom/extend-expect";

afterEach(cleanup);

describe('Main Template layout ', () => {
  test('matches the snapshot', () => {
    expect(
      render(
        <MemoryRouter>
          <MainTemplate>
            <div data-testid="mainTemplateContent">main template Content</div>
          </MainTemplate>
        </MemoryRouter>
      )
    ).toMatchSnapshot();
  });

  test('passes the children components correctly', () => {
    const {getByTestId} = render(
      <MemoryRouter>
        <MainTemplate>
          <div data-testid="mainTemplateContent">main template Content</div>
        </MainTemplate>
      </MemoryRouter>
    )
    expect(getByTestId("mainTemplateContent")).toBeInTheDocument();
  });

  test('drawer is opened by default', async () => {
    const getById = queryByAttribute.bind(null, 'id');
    const dom = render(
      <MemoryRouter>
        <MainTemplate>
          <div data-testid="mainTemplateContent">main template Content</div>
        </MainTemplate>
      </MemoryRouter>
    );

    const pageSideBar = getById(dom.container, 'page-sidebar');
    expect(pageSideBar).toHaveClass('pf-c-page__sidebar pf-m-collapsed');
  });

  test('toggle the drawer between opened and closed', async () => {
    const getById = queryByAttribute.bind(null, 'id');
    const dom = render(
      <MemoryRouter>
        <MainTemplate>
          <div data-testid="mainTemplateContent">main template Content</div>
        </MainTemplate>
      </MemoryRouter>
    );

    const navToggle = getById(dom.container, 'nav-toggle');
    await act(async () => {
      await fireEvent.click(navToggle as Element);
      const pageSideBar = getById(dom.container, 'page-sidebar');
      expect(pageSideBar).toHaveClass('pf-c-page__sidebar pf-m-expanded');
    })
  })
});




