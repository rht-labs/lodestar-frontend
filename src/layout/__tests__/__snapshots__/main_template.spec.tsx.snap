// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main Template layout  matches the snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="pf-c-page"
        style="height: 100vh;"
      >
        <header
          class="pf-c-page__header"
        >
          <div
            class="pf-c-page__header-brand"
          >
            <div
              class="pf-c-page__header-brand-toggle"
            >
              <button
                aria-controls="page-sidebar"
                aria-expanded="true"
                aria-label="Global navigation"
                class="pf-c-button pf-m-plain"
                data-ouia-component-type="PF4/Button"
                data-ouia-safe="true"
                id="nav-toggle"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  fill="currentColor"
                  height="1em"
                  role="img"
                  style="vertical-align: -0.125em;"
                  viewBox="0 0 448 512"
                  width="1em"
                >
                  <path
                    d="M16 132h416c8.837 0 16-7.163 16-16V76c0-8.837-7.163-16-16-16H16C7.163 60 0 67.163 0 76v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16z"
                    transform=""
                  />
                </svg>
              </button>
            </div>
            <a
              class="pf-c-page__header-brand-link"
            >
              <img
                alt="Open Innovation Labs"
                class="pf-c-brand"
                src="/oil_logo.png"
              />
            </a>
          </div>
          <div
            class="pf-c-page__header-tools"
          >
            <div
              class="pf-c-page__header-tools-group"
            >
              <div
                class="pf-c-page__header-tools-item"
              >
                <button
                  aria-label="Notifications"
                  class="pf-c-button pf-m-plain"
                  data-ouia-component-type="PF4/Button"
                  data-ouia-safe="true"
                  type="button"
                >
                  <span
                    class="pf-c-notification-badge pf-m-read"
                  >
                    <svg
                      aria-hidden="true"
                      fill="currentColor"
                      height="1em"
                      role="img"
                      style="vertical-align: -0.125em;"
                      viewBox="0 0 448 512"
                      width="1em"
                    >
                      <path
                        d="M224 512c35.32 0 63.97-28.65 63.97-64H160.03c0 35.35 28.65 64 63.97 64zm215.39-149.71c-19.32-20.76-55.47-51.99-55.47-154.29 0-77.7-54.48-139.9-127.94-155.16V32c0-17.67-14.32-32-31.98-32s-31.98 14.33-31.98 32v20.84C118.56 68.1 64.08 130.3 64.08 208c0 102.3-36.15 133.53-55.47 154.29-6 6.45-8.66 14.16-8.61 21.71.11 16.4 12.98 32 32.1 32h383.8c19.12 0 32-15.6 32.1-32 .05-7.55-2.61-15.27-8.61-21.71z"
                        transform=""
                      />
                    </svg>
                  </span>
                </button>
              </div>
              <div
                class="pf-c-page__header-tools-item"
              >
                <nav
                  aria-label="Global"
                  class="pf-c-nav pf-m-horizontal"
                  data-ouia-component-id="3"
                  data-ouia-component-type="PF4/Nav"
                  data-ouia-safe="true"
                >
                  <button
                    aria-label="Scroll left"
                    class="pf-c-nav__scroll-button"
                    disabled=""
                  >
                    <svg
                      aria-hidden="true"
                      fill="currentColor"
                      height="1em"
                      role="img"
                      style="vertical-align: -0.125em;"
                      viewBox="0 0 256 512"
                      width="1em"
                    >
                      <path
                        d="M31.7 239l136-136c9.4-9.4 24.6-9.4 33.9 0l22.6 22.6c9.4 9.4 9.4 24.6 0 33.9L127.9 256l96.4 96.4c9.4 9.4 9.4 24.6 0 33.9L201.7 409c-9.4 9.4-24.6 9.4-33.9 0l-136-136c-9.5-9.4-9.5-24.6-.1-34z"
                        transform=""
                      />
                    </svg>
                  </button>
                  <ul
                    class="pf-c-nav__list"
                  >
                    <li
                      class="pf-c-nav__item"
                    >
                      <a
                        class="pf-c-nav__link"
                        href="/app/about"
                        style="padding-bottom: 0px; padding-top: 0px;"
                      >
                        <svg
                          aria-labelledby="icon-title-1"
                          fill="currentColor"
                          height="1em"
                          role="img"
                          style="vertical-align: -0.125em;"
                          viewBox="0 64 1024 1024"
                          width="1em"
                        >
                          <title
                            id="icon-title-1"
                          >
                            About
                          </title>
                          <path
                            d="M512.059-73.143c-282.338 0-512.059 229.673-512.059 512.025 0 282.235 229.721 511.975 512.059 511.975 282.281 0 511.941-229.735 511.941-511.975 0.005-282.352-229.659-512.025-511.941-512.025zM512.059 826.523c-213.826 0-387.728-173.856-387.728-387.643 0-213.89 173.904-387.694 387.728-387.694 213.717 0 387.671 173.803 387.671 387.694 0.005 213.785-173.957 387.643-387.671 387.643zM585.143 164.571v109.714c0 4.951-1.808 9.237-5.429 12.857s-7.906 5.429-12.857 5.429h-109.714c-4.953 0-9.239-1.808-12.857-5.429s-5.429-7.906-5.429-12.857v-109.714c0-4.951 1.81-9.237 5.429-12.857s7.904-5.429 12.857-5.429h109.714c4.951 0 9.237 1.808 12.857 5.429s5.429 7.906 5.429 12.857zM521.616 365.714c118.343 0 214.286 88.965 214.286 187.429s-95.943 178.286-214.286 178.286c-173.045 0-208.091-93.714-213.963-171.113 0-7.936 6.729-9.838 15.872-9.838s108.073 0 113.143 0c6.096 0 14.475 0.633 17.554 10.571 0 48.857 135.968 54.953 135.968-7.904 0-31.506-29.717-63.817-68.571-66.194s-82.286-7.607-82.286-54.199c0-13.022 0-25.673 0-44.693 0-19.015 9.717-22.343 27.431-22.343s54.853-0.002 54.853-0.002z"
                            transform="rotate(180 0 512) scale(-1 1)"
                          />
                        </svg>
                      </a>
                    </li>
                  </ul>
                  <button
                    aria-label="Scroll right"
                    class="pf-c-nav__scroll-button"
                    disabled=""
                  >
                    <svg
                      aria-hidden="true"
                      fill="currentColor"
                      height="1em"
                      role="img"
                      style="vertical-align: -0.125em;"
                      viewBox="0 0 256 512"
                      width="1em"
                    >
                      <path
                        d="M224.3 273l-136 136c-9.4 9.4-24.6 9.4-33.9 0l-22.6-22.6c-9.4-9.4-9.4-24.6 0-33.9l96.4-96.4-96.4-96.4c-9.4-9.4-9.4-24.6 0-33.9L54.3 103c9.4-9.4 24.6-9.4 33.9 0l136 136c9.5 9.4 9.5 24.6.1 34z"
                        transform=""
                      />
                    </svg>
                  </button>
                </nav>
              </div>
              <div
                class="pf-c-page__header-tools-item"
              >
                <div
                  style="width: auto; height: 50px; justify-content: center; display: flex; align-items: center;"
                >
                  <div
                    class="pf-c-dropdown"
                    data-ouia-component-id="1"
                    data-ouia-component-type="PF4/Dropdown"
                    data-ouia-safe="true"
                  >
                    <button
                      aria-expanded="false"
                      aria-haspopup="true"
                      class="pf-c-dropdown__toggle pf-m-plain"
                      id="pf-toggle-id-0"
                      type="button"
                    >
                      <span
                        class="pf-c-dropdown__toggle-icon"
                      >
                        <svg
                          aria-hidden="true"
                          fill="currentColor"
                          height="1em"
                          role="img"
                          style="vertical-align: -0.125em;"
                          viewBox="0 0 320 512"
                          width="1em"
                        >
                          <path
                            d="M31.3 192h257.3c17.8 0 26.7 21.5 14.1 34.1L174.1 354.8c-7.8 7.8-20.5 7.8-28.3 0L17.2 226.1C4.6 213.5 13.5 192 31.3 192z"
                            transform=""
                          />
                        </svg>
                      </span>
                    </button>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </header>
        <div
          class="pf-c-page__sidebar pf-m-expanded"
          id="page-sidebar"
        >
          <div
            class="pf-c-page__sidebar-body"
          >
            <div>
              <nav
                aria-label="Global"
                class="pf-c-nav"
                data-ouia-component-id="4"
                data-ouia-component-type="PF4/Nav"
                data-ouia-safe="true"
              >
                <li
                  class="pf-c-nav__item"
                >
                  <a
                    class="pf-c-nav__link"
                    href="/app/dashboard"
                  >
                    Dashboard
                  </a>
                </li>
                <ul
                  class="pf-c-nav__list"
                >
                  <li
                    class="pf-c-nav__item pf-m-expanded"
                  >
                    <a
                      aria-expanded="true"
                      class="pf-c-nav__link"
                      href="#"
                      id="navExpandableEngagement"
                    >
                      Engagements
                      <span
                        class="pf-c-nav__toggle"
                      >
                        <span
                          class="pf-c-nav__toggle-icon"
                        >
                          <svg
                            aria-hidden="true"
                            fill="currentColor"
                            height="1em"
                            role="img"
                            style="vertical-align: -0.125em;"
                            viewBox="0 0 256 512"
                            width="1em"
                          >
                            <path
                              d="M224.3 273l-136 136c-9.4 9.4-24.6 9.4-33.9 0l-22.6-22.6c-9.4-9.4-9.4-24.6 0-33.9l96.4-96.4-96.4-96.4c-9.4-9.4-9.4-24.6 0-33.9L54.3 103c9.4-9.4 24.6-9.4 33.9 0l136 136c9.5 9.4 9.5 24.6.1 34z"
                              transform=""
                            />
                          </svg>
                        </span>
                      </span>
                    </a>
                    <section
                      aria-labelledby="navExpandableEngagement"
                      class="pf-c-nav__subnav"
                    >
                      
                      <ul
                        class="pf-c-nav__list"
                      >
                        <li
                          class="pf-c-nav__item"
                        >
                          <a
                            class="pf-c-nav__link"
                            href="/app/engagements/all"
                          >
                            Overview
                          </a>
                        </li>
                        <li
                          class="pf-c-nav__item"
                        >
                          <a
                            class="pf-c-nav__link"
                            href="/app/engagements/upcoming"
                          >
                            Upcoming
                          </a>
                        </li>
                        <li
                          class="pf-c-nav__item"
                        >
                          <a
                            class="pf-c-nav__link"
                            href="/app/engagements/active"
                          >
                            Active
                          </a>
                        </li>
                        <li
                          class="pf-c-nav__item"
                        >
                          <a
                            class="pf-c-nav__link"
                            href="/app/engagements/past"
                          >
                            Past
                          </a>
                        </li>
                        <li
                          class="pf-c-nav__item"
                        >
                          <a
                            class="pf-c-nav__link"
                            href="/app/engagements/new"
                          >
                            + Create New
                          </a>
                        </li>
                      </ul>
                    </section>
                  </li>
                </ul>
              </nav>
            </div>
          </div>
        </div>
        <main
          class="pf-c-page__main"
          tabindex="-1"
        >
          <div
            class="pf-c-drawer"
          >
            <div
              class="pf-c-drawer__main"
            >
              <div
                class="pf-c-drawer__content"
              >
                <div
                  data-testid="mainTemplateContent"
                >
                  main template Content
                </div>
              </div>
              <div
                class="pf-c-drawer__panel"
                hidden=""
              >
                <div
                  class="pf-c-drawer__body"
                >
                  <div
                    class="pf-c-drawer__head"
                  >
                    <p
                      class=""
                      data-pf-content="true"
                    >
                      <b>
                         All Notifications 
                      </b>
                    </p>
                    <div
                      class="pf-c-drawer__actions"
                    >
                      <div
                        class="pf-c-drawer__close"
                      >
                        <button
                          aria-label="Close drawer panel"
                          class="pf-c-button pf-m-plain"
                          data-ouia-component-type="PF4/Button"
                          data-ouia-safe="true"
                          type="button"
                        >
                          <svg
                            aria-hidden="true"
                            fill="currentColor"
                            height="1em"
                            role="img"
                            style="vertical-align: -0.125em;"
                            viewBox="0 0 352 512"
                            width="1em"
                          >
                            <path
                              d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
                              transform=""
                            />
                          </svg>
                        </button>
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  class="pf-c-empty-state"
                >
                  <div
                    class="pf-c-empty-state__content"
                  >
                    <svg
                      aria-hidden="true"
                      class="pf-c-empty-state__icon"
                      fill="currentColor"
                      height="1em"
                      role="img"
                      style="vertical-align: -0.125em;"
                      viewBox="0 0 448 512"
                      width="1em"
                    >
                      <path
                        d="M224 512c35.32 0 63.97-28.65 63.97-64H160.03c0 35.35 28.65 64 63.97 64zm215.39-149.71c-19.32-20.76-55.47-51.99-55.47-154.29 0-77.7-54.48-139.9-127.94-155.16V32c0-17.67-14.32-32-31.98-32s-31.98 14.33-31.98 32v20.84C118.56 68.1 64.08 130.3 64.08 208c0 102.3-36.15 133.53-55.47 154.29-6 6.45-8.66 14.16-8.61 21.71.11 16.4 12.98 32 32.1 32h383.8c19.12 0 32-15.6 32.1-32 .05-7.55-2.61-15.27-8.61-21.71z"
                        transform=""
                      />
                    </svg>
                    <h4
                      class="pf-c-title pf-m-lg"
                    >
                      No new notifications
                    </h4>
                    <div
                      class="pf-c-empty-state__body"
                    >
                      No new notifications are available. Please come back later.
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </main>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="pf-c-page"
      style="height: 100vh;"
    >
      <header
        class="pf-c-page__header"
      >
        <div
          class="pf-c-page__header-brand"
        >
          <div
            class="pf-c-page__header-brand-toggle"
          >
            <button
              aria-controls="page-sidebar"
              aria-expanded="true"
              aria-label="Global navigation"
              class="pf-c-button pf-m-plain"
              data-ouia-component-type="PF4/Button"
              data-ouia-safe="true"
              id="nav-toggle"
              type="button"
            >
              <svg
                aria-hidden="true"
                fill="currentColor"
                height="1em"
                role="img"
                style="vertical-align: -0.125em;"
                viewBox="0 0 448 512"
                width="1em"
              >
                <path
                  d="M16 132h416c8.837 0 16-7.163 16-16V76c0-8.837-7.163-16-16-16H16C7.163 60 0 67.163 0 76v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16z"
                  transform=""
                />
              </svg>
            </button>
          </div>
          <a
            class="pf-c-page__header-brand-link"
          >
            <img
              alt="Open Innovation Labs"
              class="pf-c-brand"
              src="/oil_logo.png"
            />
          </a>
        </div>
        <div
          class="pf-c-page__header-tools"
        >
          <div
            class="pf-c-page__header-tools-group"
          >
            <div
              class="pf-c-page__header-tools-item"
            >
              <button
                aria-label="Notifications"
                class="pf-c-button pf-m-plain"
                data-ouia-component-type="PF4/Button"
                data-ouia-safe="true"
                type="button"
              >
                <span
                  class="pf-c-notification-badge pf-m-read"
                >
                  <svg
                    aria-hidden="true"
                    fill="currentColor"
                    height="1em"
                    role="img"
                    style="vertical-align: -0.125em;"
                    viewBox="0 0 448 512"
                    width="1em"
                  >
                    <path
                      d="M224 512c35.32 0 63.97-28.65 63.97-64H160.03c0 35.35 28.65 64 63.97 64zm215.39-149.71c-19.32-20.76-55.47-51.99-55.47-154.29 0-77.7-54.48-139.9-127.94-155.16V32c0-17.67-14.32-32-31.98-32s-31.98 14.33-31.98 32v20.84C118.56 68.1 64.08 130.3 64.08 208c0 102.3-36.15 133.53-55.47 154.29-6 6.45-8.66 14.16-8.61 21.71.11 16.4 12.98 32 32.1 32h383.8c19.12 0 32-15.6 32.1-32 .05-7.55-2.61-15.27-8.61-21.71z"
                      transform=""
                    />
                  </svg>
                </span>
              </button>
            </div>
            <div
              class="pf-c-page__header-tools-item"
            >
              <nav
                aria-label="Global"
                class="pf-c-nav pf-m-horizontal"
                data-ouia-component-id="3"
                data-ouia-component-type="PF4/Nav"
                data-ouia-safe="true"
              >
                <button
                  aria-label="Scroll left"
                  class="pf-c-nav__scroll-button"
                  disabled=""
                >
                  <svg
                    aria-hidden="true"
                    fill="currentColor"
                    height="1em"
                    role="img"
                    style="vertical-align: -0.125em;"
                    viewBox="0 0 256 512"
                    width="1em"
                  >
                    <path
                      d="M31.7 239l136-136c9.4-9.4 24.6-9.4 33.9 0l22.6 22.6c9.4 9.4 9.4 24.6 0 33.9L127.9 256l96.4 96.4c9.4 9.4 9.4 24.6 0 33.9L201.7 409c-9.4 9.4-24.6 9.4-33.9 0l-136-136c-9.5-9.4-9.5-24.6-.1-34z"
                      transform=""
                    />
                  </svg>
                </button>
                <ul
                  class="pf-c-nav__list"
                >
                  <li
                    class="pf-c-nav__item"
                  >
                    <a
                      class="pf-c-nav__link"
                      href="/app/about"
                      style="padding-bottom: 0px; padding-top: 0px;"
                    >
                      <svg
                        aria-labelledby="icon-title-1"
                        fill="currentColor"
                        height="1em"
                        role="img"
                        style="vertical-align: -0.125em;"
                        viewBox="0 64 1024 1024"
                        width="1em"
                      >
                        <title
                          id="icon-title-1"
                        >
                          About
                        </title>
                        <path
                          d="M512.059-73.143c-282.338 0-512.059 229.673-512.059 512.025 0 282.235 229.721 511.975 512.059 511.975 282.281 0 511.941-229.735 511.941-511.975 0.005-282.352-229.659-512.025-511.941-512.025zM512.059 826.523c-213.826 0-387.728-173.856-387.728-387.643 0-213.89 173.904-387.694 387.728-387.694 213.717 0 387.671 173.803 387.671 387.694 0.005 213.785-173.957 387.643-387.671 387.643zM585.143 164.571v109.714c0 4.951-1.808 9.237-5.429 12.857s-7.906 5.429-12.857 5.429h-109.714c-4.953 0-9.239-1.808-12.857-5.429s-5.429-7.906-5.429-12.857v-109.714c0-4.951 1.81-9.237 5.429-12.857s7.904-5.429 12.857-5.429h109.714c4.951 0 9.237 1.808 12.857 5.429s5.429 7.906 5.429 12.857zM521.616 365.714c118.343 0 214.286 88.965 214.286 187.429s-95.943 178.286-214.286 178.286c-173.045 0-208.091-93.714-213.963-171.113 0-7.936 6.729-9.838 15.872-9.838s108.073 0 113.143 0c6.096 0 14.475 0.633 17.554 10.571 0 48.857 135.968 54.953 135.968-7.904 0-31.506-29.717-63.817-68.571-66.194s-82.286-7.607-82.286-54.199c0-13.022 0-25.673 0-44.693 0-19.015 9.717-22.343 27.431-22.343s54.853-0.002 54.853-0.002z"
                          transform="rotate(180 0 512) scale(-1 1)"
                        />
                      </svg>
                    </a>
                  </li>
                </ul>
                <button
                  aria-label="Scroll right"
                  class="pf-c-nav__scroll-button"
                  disabled=""
                >
                  <svg
                    aria-hidden="true"
                    fill="currentColor"
                    height="1em"
                    role="img"
                    style="vertical-align: -0.125em;"
                    viewBox="0 0 256 512"
                    width="1em"
                  >
                    <path
                      d="M224.3 273l-136 136c-9.4 9.4-24.6 9.4-33.9 0l-22.6-22.6c-9.4-9.4-9.4-24.6 0-33.9l96.4-96.4-96.4-96.4c-9.4-9.4-9.4-24.6 0-33.9L54.3 103c9.4-9.4 24.6-9.4 33.9 0l136 136c9.5 9.4 9.5 24.6.1 34z"
                      transform=""
                    />
                  </svg>
                </button>
              </nav>
            </div>
            <div
              class="pf-c-page__header-tools-item"
            >
              <div
                style="width: auto; height: 50px; justify-content: center; display: flex; align-items: center;"
              >
                <div
                  class="pf-c-dropdown"
                  data-ouia-component-id="1"
                  data-ouia-component-type="PF4/Dropdown"
                  data-ouia-safe="true"
                >
                  <button
                    aria-expanded="false"
                    aria-haspopup="true"
                    class="pf-c-dropdown__toggle pf-m-plain"
                    id="pf-toggle-id-0"
                    type="button"
                  >
                    <span
                      class="pf-c-dropdown__toggle-icon"
                    >
                      <svg
                        aria-hidden="true"
                        fill="currentColor"
                        height="1em"
                        role="img"
                        style="vertical-align: -0.125em;"
                        viewBox="0 0 320 512"
                        width="1em"
                      >
                        <path
                          d="M31.3 192h257.3c17.8 0 26.7 21.5 14.1 34.1L174.1 354.8c-7.8 7.8-20.5 7.8-28.3 0L17.2 226.1C4.6 213.5 13.5 192 31.3 192z"
                          transform=""
                        />
                      </svg>
                    </span>
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
      </header>
      <div
        class="pf-c-page__sidebar pf-m-expanded"
        id="page-sidebar"
      >
        <div
          class="pf-c-page__sidebar-body"
        >
          <div>
            <nav
              aria-label="Global"
              class="pf-c-nav"
              data-ouia-component-id="4"
              data-ouia-component-type="PF4/Nav"
              data-ouia-safe="true"
            >
              <li
                class="pf-c-nav__item"
              >
                <a
                  class="pf-c-nav__link"
                  href="/app/dashboard"
                >
                  Dashboard
                </a>
              </li>
              <ul
                class="pf-c-nav__list"
              >
                <li
                  class="pf-c-nav__item pf-m-expanded"
                >
                  <a
                    aria-expanded="true"
                    class="pf-c-nav__link"
                    href="#"
                    id="navExpandableEngagement"
                  >
                    Engagements
                    <span
                      class="pf-c-nav__toggle"
                    >
                      <span
                        class="pf-c-nav__toggle-icon"
                      >
                        <svg
                          aria-hidden="true"
                          fill="currentColor"
                          height="1em"
                          role="img"
                          style="vertical-align: -0.125em;"
                          viewBox="0 0 256 512"
                          width="1em"
                        >
                          <path
                            d="M224.3 273l-136 136c-9.4 9.4-24.6 9.4-33.9 0l-22.6-22.6c-9.4-9.4-9.4-24.6 0-33.9l96.4-96.4-96.4-96.4c-9.4-9.4-9.4-24.6 0-33.9L54.3 103c9.4-9.4 24.6-9.4 33.9 0l136 136c9.5 9.4 9.5 24.6.1 34z"
                            transform=""
                          />
                        </svg>
                      </span>
                    </span>
                  </a>
                  <section
                    aria-labelledby="navExpandableEngagement"
                    class="pf-c-nav__subnav"
                  >
                    
                    <ul
                      class="pf-c-nav__list"
                    >
                      <li
                        class="pf-c-nav__item"
                      >
                        <a
                          class="pf-c-nav__link"
                          href="/app/engagements/all"
                        >
                          Overview
                        </a>
                      </li>
                      <li
                        class="pf-c-nav__item"
                      >
                        <a
                          class="pf-c-nav__link"
                          href="/app/engagements/upcoming"
                        >
                          Upcoming
                        </a>
                      </li>
                      <li
                        class="pf-c-nav__item"
                      >
                        <a
                          class="pf-c-nav__link"
                          href="/app/engagements/active"
                        >
                          Active
                        </a>
                      </li>
                      <li
                        class="pf-c-nav__item"
                      >
                        <a
                          class="pf-c-nav__link"
                          href="/app/engagements/past"
                        >
                          Past
                        </a>
                      </li>
                      <li
                        class="pf-c-nav__item"
                      >
                        <a
                          class="pf-c-nav__link"
                          href="/app/engagements/new"
                        >
                          + Create New
                        </a>
                      </li>
                    </ul>
                  </section>
                </li>
              </ul>
            </nav>
          </div>
        </div>
      </div>
      <main
        class="pf-c-page__main"
        tabindex="-1"
      >
        <div
          class="pf-c-drawer"
        >
          <div
            class="pf-c-drawer__main"
          >
            <div
              class="pf-c-drawer__content"
            >
              <div
                data-testid="mainTemplateContent"
              >
                main template Content
              </div>
            </div>
            <div
              class="pf-c-drawer__panel"
              hidden=""
            >
              <div
                class="pf-c-drawer__body"
              >
                <div
                  class="pf-c-drawer__head"
                >
                  <p
                    class=""
                    data-pf-content="true"
                  >
                    <b>
                       All Notifications 
                    </b>
                  </p>
                  <div
                    class="pf-c-drawer__actions"
                  >
                    <div
                      class="pf-c-drawer__close"
                    >
                      <button
                        aria-label="Close drawer panel"
                        class="pf-c-button pf-m-plain"
                        data-ouia-component-type="PF4/Button"
                        data-ouia-safe="true"
                        type="button"
                      >
                        <svg
                          aria-hidden="true"
                          fill="currentColor"
                          height="1em"
                          role="img"
                          style="vertical-align: -0.125em;"
                          viewBox="0 0 352 512"
                          width="1em"
                        >
                          <path
                            d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
                            transform=""
                          />
                        </svg>
                      </button>
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="pf-c-empty-state"
              >
                <div
                  class="pf-c-empty-state__content"
                >
                  <svg
                    aria-hidden="true"
                    class="pf-c-empty-state__icon"
                    fill="currentColor"
                    height="1em"
                    role="img"
                    style="vertical-align: -0.125em;"
                    viewBox="0 0 448 512"
                    width="1em"
                  >
                    <path
                      d="M224 512c35.32 0 63.97-28.65 63.97-64H160.03c0 35.35 28.65 64 63.97 64zm215.39-149.71c-19.32-20.76-55.47-51.99-55.47-154.29 0-77.7-54.48-139.9-127.94-155.16V32c0-17.67-14.32-32-31.98-32s-31.98 14.33-31.98 32v20.84C118.56 68.1 64.08 130.3 64.08 208c0 102.3-36.15 133.53-55.47 154.29-6 6.45-8.66 14.16-8.61 21.71.11 16.4 12.98 32 32.1 32h383.8c19.12 0 32-15.6 32.1-32 .05-7.55-2.61-15.27-8.61-21.71z"
                      transform=""
                    />
                  </svg>
                  <h4
                    class="pf-c-title pf-m-lg"
                  >
                    No new notifications
                  </h4>
                  <div
                    class="pf-c-empty-state__body"
                  >
                    No new notifications are available. Please come back later.
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </main>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
